### ISSUE TITLE
Creating Kafka topic with standalone Topic Operator

### ISSUE THREAD

**Issue by Ihjaz** [strimzi/strimzi-kafka-operator#971]
Hi,

What should be the label used when creating a kafka topic with standalone Topic Operator?

If I use the name of my kubernetes cluster which is not created using Cluster Operator then topic operator is not picking it up.

```
apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaTopic
metadata:
  name: mytopic
  labels:
    strimzi.io/cluster: kubernetes
```
I saw that the label used for topics created outside the operator is "strimzi.io/kind: topic".
If I use the label as "strimzi.io/kind: topic" then topic operator is getting notified and creating the topic but I see the topic operator pod restarting when I delete the topic.

```
apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaTopic
metadata:
  name: mytopic
  labels:
    strimzi.io/kind: topic
```

Is the Topic operator restarting because I'm using this label ? What is the right label to use if the cluster is not created using a Cluster Operator?

**Comment by scholzj** [comment]
Hi,

You should be able to set the labels using the environment variable `STRIMZI_RESOURCE_LABELS` where you can specify the labels which should be used on the KafkaTopic resource to be picked up. I would expect that by default it should be empty. 

Sadly, it seems that the deployment configuration is not up-to-date and has still the old variable `STRIMZI_CONFIGMAP_LABELS` which should not work in the last versions of Strimzi.

It definitely shouldn't restart after deleting the topic - could you share the log so that we can dig into the restarts?

@tombentley Do you know a bit more about the labels configuration? You understand TO much better than I do.

**Comment by tombentley** [comment]
On Thu, Oct 11, 2018 at 9:30 PM Jakub Scholz <notifications@github.com>
wrote:

> Hi,
>
> You should be able to set the labels using the environment variable
> STRIMZI_RESOURCE_LABELS where you can specify the labels which should be
> used on the KafkaTopic resource to be picked up. I would expect that by
> default it should be empty.
>

The default seems to be strimzi.io/kind=topic, which really is a hang-over
from when this operator was consuming configmaps. We should fix that as
well as the deployment config. I've opened
https://github.com/strimzi/strimzi-kafka-operator/issues/974 for this.


> Sadly, it seems that the deployment configuration is not up-to-date and
> has still the old variable STRIMZI_CONFIGMAP_LABELS which should not work
> in the last versions of Strimzi.
>
> It definitely shouldn't restart after deleting the topic - could you share
> the log so that we can dig into the restarts?
>
I agree these restarts sound like a different problem.

**Comment by Ihjaz** [comment]
Hi,

It seems like pod restarts were due to some connectivity issue between the nodes on my cluster. I'm not seeing it anymore.

Also please consider updating the document with a sample KafkaTopic definition with appropriate label that can be used with a standalone TopicOperator deployment.

**Comment by scholzj** [comment]
This should be addressed by #1430
